src/xplatform/set.rs

```diff
@@ -117,6 +117,41 @@ impl Peer {
             allowed_ips: vec![],
         }
     }
+
+    pub fn remove(mut self, remove: bool) -> Self {
+        self.remove = remove.then(|| true);
+        self
+    }
+
+    pub fn update_only(mut self, update_only: bool) -> Self {
+        self.update_only = update_only.then(|| true);
+        self
+    }
+
+    pub fn preshared_key(mut self, preshared_key: [u8; 32]) -> Self {
+        self.preshared_key = Some(preshared_key);
+        self
+    }
+
+    pub fn endpoint(mut self, endpoint: SocketAddr) -> Self {
+        self.endpoint = Some(endpoint);
+        self
+    }
+
+    pub fn persistent_keepalive_interval(mut self, persistent_keepalive_interval: u16) -> Self {
+        self.persistent_keepalive_interval = Some(persistent_keepalive_interval);
```

Comment:
Nitpick that this API doesn't allow you to unset `persistent_keepalive_interval`, but not a big deal since fields are public.